servers/visual/visual_server_viewport.cpp

```diff
@@ -419,21 +420,40 @@ void VisualServerViewport::viewport_set_clear_mode(RID p_viewport, VS::ViewportC
 	viewport->clear_mode = p_clear_mode;
 }
 
-void VisualServerViewport::viewport_attach_to_screen(RID p_viewport, const Rect2 &p_rect, int p_screen) {
+void VisualServerViewport::viewport_attach_to_screen(RID p_viewport, const Rect2 &p_rect, int p_screen, bool p_optimize) {
 
 	Viewport *viewport = viewport_owner.getornull(p_viewport);
 	ERR_FAIL_COND(!viewport);
 
+	// If using GLES2 we optimize this operation by rendering directly to system_fbo
+	// instead of rendering to fbo and copying to system_fbo after
+	if (VSG::rasterizer->is_low_end() && p_optimize) {
+		//free the current render_target
+		VSG::storage->free(viewport->render_target);
+		//create a new render_target with system fbo
+		viewport->render_target = VSG::storage->render_target_screen_create(p_rect);
+	}
+
 	viewport->viewport_to_screen_rect = p_rect;
 	viewport->viewport_to_screen = p_screen;
+	viewport->viewport_use_screen_optimization = p_optimize;
```

Comment:
name should be
use_render_direct_to_screen instead of use_screen_optimization, so it's more obvious what it does.